### PR TITLE
Rotation, finally

### LINKED ISSUE #131: It would be nice to be able to resize the image with a mouse scroll movement when pasting an image with Stamp image or Ctrl + v.
It would be nice to be able to resize the image with a mouse scroll movement when pasting an image with Stamp image or Ctrl + v. 
Currently, it is necessary to resize using Select image additionally, and there is a problem of selecting other images on the same layer together.
And is it possible to rotate the image as well?

### PR DESCRIPTION
**Pull request by seijihariki**
Finally implements rotation to select tool

Will use this same interface for stamp tool to allow for scaling/rotation in this same pull request.

This pull aims to resolve #131 and #156.

### CODE REVIEW

**Comment by seijihariki** [comment]
Needs testing with chrome.

**Comment by zero01101** [comment]
this is absolutely _fantastic_ work; i've actually only used it in chrome so far and it's performing marvelously <3

my apologies for being somewhat absent lately, some offline bonkersness is demanding my attention :/ thank you again for everything you do 

i'll try to give this a more thorough check-over later this evening, but initial quick glance says it's magnificent like usual :D

**Comment by seijihariki** [comment]
Gonna say. Transformations are cool and simple until you have to mix 3 different coordinate spaces across 2 different rendered viewports.

File: js/ui/tool/select.js [hunk 1]
```diff
@@ -1,3 +1,7 @@
+/**
+ * TODO: REFACTOR THIS WHOLE THING
+ */
+
```
Comment (zero01101) [inline js/ui/tool/select.js:4]: lmao

**Review by zero01101 — APPROVED** [review]
i've been trying to use both firefox and chrome for playing with this and it seems to be working brilliantly on both browsers; angle-snapping is a _godsend_ for rotation as well <3  

didn't mark generic.js and select.js as viewed because let's face it, you're smarter than i am and i wouldn't be able to understand those properly for like a year :D  

completely aside, i did notice in chrome that the giant style checkmark has re-appeared, and the magic canvas expando buttons don't display their chevron icon, and a few of the checkmark toggle button icons seem to be scaled a bit weirdly?  100% not relevant to this PR of course and simple visual quirks, just wanted to mention before ADHD made me forget about them forever lol 

managed to catch all 3 in a single screenshot, how bout that 

image

File: js/ui/tool/select.js (hunk 1 above)
Comment (seijihariki) [reply js/ui/tool/select.js:4]: Oh, that was before the latest versions of the new select. We can just remove it now.

**Comment by seijihariki** [comment]
Oh no. The gigantic check is back? That's... Very weird. Will give this a look anyways.

Will try to fix these and merge to main!

**Comment by seijihariki** [comment]
Fixed icon (and recursion) problems. Will be merging the branch.

**Comment by zero01101** [comment]
confirming magnificence <3

image

**Pull request merged**